### ISSUE TITLE
How to restore last set brightness after reboot?

### ISSUE THREAD

**Issue by apoorv569** [Hummer12007/brightnessctl#104]
How do I restore the last set brightness level after I reboot my system? As `brightnessctl` seems to save the current value under `/run/user/1000/brightnessctl/`, which doesn't persist upon a reboot.

**Comment by plum** [comment]
> Kind of a hack, but you can reset XDG_RUNTIME_DIR environmental variable when invoking brightnessctl for save/restore actions

Thanks for kind advice. I tried to understand your solution, but the man pages offer little help here. Could you explain what you mean here, please;

1. Exactly what you mean should be done to 'reset XDG_RUNTIME_DIR', AND, 
2.  What you mean by ' invoking brightnessctl for save restore' .. could it be the -s option , for save sate in temporary file?

**Comment by CameronNemo** [comment]
Yes there is the save/restore options, -s and -r. If you don't know what XDG_RUNTIME_DIR is, you probably shouldn't be messing with it. The environment variable is an XDG standard used by many pieces of software. If you do end up modifying it, make sure the changes are only scoped to brightnessctl, and only for the limited operations you need to change the variable for. Basically XDG_RUNTIME_DIR is a per-user directory set up when you login (it might be `/run/user/1000`, where 1000 is the uid of your user), it is not normally supposed to change. brightnessctl puts a file in there for the save/restore actions. It gets cleared on reboot. But if you change the XDG_RUNTIME_DIR when running brightnessctl -s/-r, then you can set the directory to something that persists across reboots.

E.g. on shutdown/session termination: `env XDG_RUNTIME_DIR=$HOME/.cache brightnessctl -s`

Then on boot to restore: `env XDG_RUNTIME_DIR=$HOME/.cache brightnessctl -r`

**Comment by plum** [comment]
@CameronNemo  THanks. That's clear now. Yes I understand the implications of changing the xdg-runtime directory, and would not do that.
However,  I can look into how to change it for brightnessctl only, which i do not know precisely how to do - some kind of drop-in in the appropriate  /etc directory, I suppose.
Regards, John
